internal/app/api/v1/testtriggers.go

```diff
@@ -96,6 +99,76 @@ func (s *TestkubeAPI) UpdateTestTriggerHandler() fiber.Handler {
 	}
 }
 
+// BulkUpdateTestTriggersHandler is a handler for bukl updates an existing TestTrigger CRDs based on array of TestTrigger content
+func (s *TestkubeAPI) BulkUpdateTestTriggersHandler() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		var request []testkube.TestTriggerUpsertRequest
+		err := c.BodyParser(&request)
+		if err != nil {
+			return s.Error(c, http.StatusBadRequest, err)
+		}
+
+		if request == nil || len(request) == 0 {
+			return s.Error(c, http.StatusBadRequest, errors.New("invalid request body: expected array of test trigger upsert requests"))
+		}
+
+		namespace := s.Namespace
+
+		testTriggers := make([]testkube.TestTrigger, 0)
+
+		for _, upsertRequest := range request {
+			if upsertRequest.Namespace != "" {
+				namespace = upsertRequest.Namespace
+			}
+			var testTrigger *testtriggersv1.TestTrigger
+			crdTestTrigger := testtriggersmapper.MapTestTriggerUpsertRequestToTestTriggerCRD(upsertRequest)
+			// we need to get resource first and load its metadata.ResourceVersion
+			testTrigger, err := s.TestKubeClientset.
+				TestsV1().
+				TestTriggers(namespace).
+				Get(c.UserContext(), upsertRequest.Name, v1.GetOptions{})
+			if k8serrors.IsNotFound(err) {
+				testTrigger, err = s.TestKubeClientset.
+					TestsV1().
+					TestTriggers(namespace).
+					Create(c.UserContext(), &crdTestTrigger, v1.CreateOptions{})
+
+				s.Metrics.IncCreateTestTrigger(err)
+
+				if err != nil {
+					err := errors.Wrapf(err, "error creating new test trigger %s/%s", testTrigger.Namespace, testTrigger.Name)
+					return s.Error(c, http.StatusBadGateway, err)
+				}
+			} else if err != nil {
+				return s.Error(c, http.StatusBadGateway, err)
+			} else {
+				// map TestSuite but load spec only to not override metadata.ResourceVersion
+				testTrigger.Spec = crdTestTrigger.Spec
+				testTrigger.Labels = upsertRequest.Labels
+				testTrigger, err = s.TestKubeClientset.
+					TestsV1().
+					TestTriggers(namespace).
+					Update(c.UserContext(), testTrigger, v1.UpdateOptions{})
+
+				s.Metrics.IncUpdateTestTrigger(err)
+
+				if err != nil {
+					err := errors.Wrapf(err, "error updating test trigger %s/%s", testTrigger.Namespace, testTrigger.Name)
+					return s.Error(c, http.StatusBadGateway, err)
+				}
+			}
+			testTriggers = append(testTriggers, testtriggersmapper.MapCRDToAPI(testTrigger))
+		}
```

Comment:
what about deleted test triggers?

Reply:
this endpoint can only be used for bulk update and insert operations, delete trigger is a separate endpoint

Reply:
I  don't know zn original Pavlo request, but I can think, that he just wants to send an array of test triggers (added, updated or deleted) and api should just sync this array with existing crds

Reply:
@vsukhin @dejanzele that is true, I can even send an empty array, then every trigger should be removed

Reply:
It will work, of course, if we delete all, then create all, but we might use something a bit more clever like I did for
https://github.com/kubeshop/testkube/blob/main/internal/app/api/v1/testsource.go#L153